tango_simlib/fandango_json_parser.py

```diff
@@ -0,0 +1,193 @@
+######################################################################################### 
+# Author: cam@ska.ac.za                                                                 #
+# Copyright 2018 SKA South Africa (http://ska.ac.za/)                                   #
+#                                                                                       #
+# BSD license - see LICENSE.txt for details                                             #
+#########################################################################################
+"""This module performs the parsing of the TANGO device data json file, containing the
+information needed to instantiate a useful device simulator.
+Instructions on how to create this json file can be found at the link below:
+https://github.com/tango-controls/fandango/blob/master/doc/recipes/ExportDeviceData.rst
+"""
+
+import logging
+import json
+
+from helper_module import json_load_byteified
+
+from PyTango._PyTango import CmdArgType, AttrDataFormat
+
+MODULE_LOGGER = logging.getLogger(__name__)
+
+CMD_PROP_MAP = {
+    'name': 'name',
+    'in_type': 'dtype_in',
+    'in_type_desc': 'doc_in',
+    'out_type': 'dtype_out',
+    'out_type_desc': 'doc_out'
+}
+
+class FandangoExportDeviceParser(object):
+
+    def __init__(self):
+        """Parser class handling a TANGO device data file in json format.
+
+        Creating an instance of this class requires calling :meth:`parse` method
+        afterwards to extract all the provided TANGO attributes, commands,
+        device property and device override class information from the specified
+        file. The formated data is a dict structure and can be obtained using
+        :meth:`get_reformatted_device_attr_metadata`,
+        :meth:`get_reformatted_cmd_metadata`,
+        :meth:`get_reformatted_properties_metadata` and
+        :meth:`get_reformatted_override_metadata`.
+        """
+
+        # Simulator decription datafile in json format
+        self.data_description_file_name = ''
+        self.device_class_name = ''
+        self._device_attributes = {}
+        """The data structure format is a dict containing attribute info in a dict.
```

Comment:
I am in two minds about these very good, but long comments in the code section. May be move to the bottom of the doc string e.g.
"""
...

self._device_attributes - The data structure format .... 
    e.g. 
       {
       }

self._device_commands - The data structure format
    e.g. 
       {
       }
"""

Reply:
@lvdheever I was thinking of moving them to the attributes' get methods'docstrings.

Reply:
OK